keps/sig-api-machinery/4008-crd-ratcheting/README.md

```diff
@@ -1284,6 +1386,85 @@ The field to which the CEL rule in its `x-kubernetes-validations` list must
 contain all fields referred by the CEL rule; so the CEL error logic can have to 
 be modified to also include the rule's original location.
 
+### Different Ratcheting Rule Per Value Validation
+
+An alternative to the weaker form ratcheting described in the proposal would be
+to apply the DeepEqual rule to a projection of the object being validated.
+
+i.e. MaxLength would use projection `len(obj)` since it only looks at that value,
+and if `len(oldObj) == len(newObj)`, then we would allow ratcheting.
```

Comment:
aren't we projecting too in the proposed solution with the predicate itself? I.e.:

```
before := MaxLength("hello")<=3
after := MaxLength("longer string")<=3

if !(after || before == after && !before && !after) {
 return Invalid()
}
```
vs. (your example from below)

```
beforeProj := len("hello")
afterProj := len("longer string")

before := MaxLength("hello")<=3
after := MaxLength("longer string")<=3

if !(after || beforeProj == afterProj && !before && !after) {
 return Invalid()
}
```

Reply:
Yes, that is what `Identity` refers to in the table of proposed projections. I changed the wording to explicitly refer to `smaller` projection.